src/plugins/dotplot/src/DotPlotWidget.cpp

```diff
@@ -1497,66 +1497,52 @@ void DotPlotWidget::selectNearestRepeat(const QPointF& p) {
     nearestSelecting = false;
 }
 
+float DotPlotWidget::calculateDistance(float x, float y, const DotPlotResults& r) const {
+    // find nearest point on line segment to desired point
+    // then calculate distance
+    double t = (double)(((x - r.x) + (y - r.y)) * r.len) / (2 * pow(r.len, 2));
+    t = std::min(1.0, std::max(0.0, t));
+    float nearestPointX = r.x + t * r.len;
```

Comment:
Min of double and float types. Can the result be double too?

Reply:
I made all float, because whole class operates with floats.

Reply:
Is there any reason for the t to be 'double'?
My comment is to avoid implicit (hidden) type conversion.

Reply:
Yes lets make this function an independent (static) at least. This is a very good candidate for a GeometryUtils-like class.